src/key_packages/mod.rs

```diff
@@ -211,6 +211,23 @@ impl KeyPackage {
     pub(crate) fn cipher_suite(&self) -> &Ciphersuite {
         self.cipher_suite
     }
+
+    /// Replace the init key in the `KeyPackage` with a random one and return a
+    /// `KeyPackageBundle` with the corresponding secret values
+    pub fn rekey(&self) -> KeyPackageBundle {
+        let ciphersuite = self.cipher_suite();
+        // Generate a new leaf secret and derive the key pair
+        let leaf_secret = Secret::from(get_random_vec(ciphersuite.hash_length()));
+        let leaf_node_secret = KeyPackageBundle::derive_leaf_node_secret(ciphersuite, &leaf_secret);
```

Comment:
I'd actually move all this into the `KeyPackageBundle` and have a `random` function there.

Reply:
😆 
That's where it was. Anyway, I moved it back now and fixed the `random` part.
I think it's better there anyway.